src/Components/DataHistoryExplorer/Internal/TimeSeriesViewer/Internal/TimeSeriesTable/TimeSeriesTable.tsx

```diff
@@ -222,6 +227,16 @@ const TimeSeriesTable: React.FC<ITimeSeriesTableProps> = (props) => {
         );
         setItems(adxTimeSeriesTableRows);
     }, [data]);
+    useEffect(() => {
+        if (data && !isLoading) {
+            const seriesWithNoData = timeSeriesTwinList
+                ?.filter((ts) => !data?.find((d) => d.seriesId === ts.seriesId))
+                .map((ts) => ts.seriesId);
+            if (onMissingSeriesData) {
+                onMissingSeriesData(seriesWithNoData);
```

Comment:
why do we need to do this? and do we really want to call it on every render where something changes? It has a weird feel that we need this... 🤔

Reply:
as we discussed, since in the current design we fetch data separately in Chart and Table, whenever there is change in data, or list we need to make the parent know about series with no data, we need to bubble up the missing series to the parent so that higher up sibling component (TimeSeriesBuilder) would update the list to show the warning icon with noData label.

When I implement single data fetch logic in the upcoming PR, I will not need that since I will feed all the internal components with the same data and the do not need to communicate with each other to show the consistent noData message.